# Determination - Deterministic rendering environment for music and art
# Copyright (C) 2024 Liu Hao <whiteaxe@tuta.io>
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, version 3.

# Krita forces the environment variable "QT_QPA_PLATFORM" to be set to "xcb" if
# X11 is detected. Unfortunately, the XCB backend doesn't work, so we need to
# unforce it here so we can manually set it to something else.

--- a/krita/main.cc
+++ b/krita/main.cc
@@ -241,7 +241,7 @@ extern "C" MAIN_EXPORT int MAIN_FN(int argc, char **argv)
     bool runningInKDE = !qgetenv("KDE_FULL_SESSION").isEmpty();
 
 #if defined HAVE_X11
-    qputenv("QT_QPA_PLATFORM", "xcb");
+    //qputenv("QT_QPA_PLATFORM", "xcb");
 #elif defined Q_OS_WIN
     if (!qEnvironmentVariableIsSet("QT_QPA_PLATFORM")) {
         qputenv("QT_QPA_PLATFORM", "windows:darkmode=1");
